client/consensus/epochs/src/lib.rs

```diff
@@ -518,8 +518,8 @@ where
 		let is_descendent_of = descendent_of_builder.build_is_descendent_of(None);
 
 		let predicate = |epoch: &PersistedEpochHeader<E>| match *epoch {
-			PersistedEpochHeader::Genesis(_, ref epoch_1) => slot >= epoch_1.end_slot,
-			PersistedEpochHeader::Regular(ref epoch_n) => slot >= epoch_n.end_slot,
+			PersistedEpochHeader::Genesis(ref epoch_0, _) => epoch_0.start_slot <= slot,
+			PersistedEpochHeader::Regular(ref epoch_n) => epoch_n.start_slot <= slot,
```

Comment:
This is the fix, the rest is only refactory

Reply:
Just for my understanding: we re-root the tree to the deepest node that passes the predicate, with the previous predicate this would only pass for an epoch that is already elapsed (the previous epoch). With the new predicate the only epoch that will pass the predicate is the one we're currently in.

Reply:
Yeah, it will pass for the epoch that we are in or the last one in the tree that is viable (if we skipped epochs).